packages/amplify-graphql-api-construct-tests/src/sql-datatabase-controller.ts

```diff
@@ -51,20 +54,34 @@ export interface SqlDatabaseDetails {
 export class SqlDatatabaseController {
   private databaseDetails: SqlDatabaseDetails | undefined;
   private useDataAPI: boolean;
+  private enableLocalTesting: boolean;
 
-  constructor(private readonly setupQueries: Array<string>, private readonly options: RDSConfig) {
+  constructor(private readonly setupQueries: Array<string>, private options: RDSConfig) {
     // Data API is not supported in opted-in regions
     if (options.engine === 'postgres' && isDataAPISupported(options.region)) {
       this.useDataAPI = true;
+      this.enableLocalTesting = !isCI() && getClusterIdentifier(options.region, options.engine) !== undefined;
     } else {
       this.useDataAPI = false;
     }
+
+    // If database name not manually set, provide and sanitize the config dbname
+    if (!options.dbname || options.dbname.length == 0 || this.enableLocalTesting) {
+      this.options.dbname = generateDBName().replace(/[^a-zA-Z0-9_]/g, '');
```

Comment:
as noted elsewhere, we can certainly keep this sanitization here, but we _also_ need to sanitize values before we interpolate them into SQL strings